controller/api/public/stat_summary.go

```diff
@@ -0,0 +1,334 @@
+		labels = append(labels, dstLabel)
+
+	case *pb.StatSummaryRequest_OutFromResource:
+		direction = "outbound"
+
+		srcLabel := fmt.Sprintf("dst_namespace=\"%s\", dst_%s=\"%s\"",
+			req.Resource.Spec.Namespace,
+			req.Resource.Spec.Type,
+			req.Resource.Spec.Name,
+		)
+		labels = append(labels, srcLabel)
+
+		outFromNs := req.GetOutFromResource().Namespace
+		if outFromNs == "" {
+			outFromNs = req.Resource.Spec.Namespace
+		}
+
+		labels = append(labels, promLabel("namespace", outFromNs))
+		if req.Resource.Spec.Name != "" {
+			labels = append(labels, promLabel(req.GetOutFromResource().Type, req.GetOutFromResource().Name))
+		}
+	default:
+		direction = "inbound"
+
+	}
+
+	// it's weird to check this again outside the switch, but including this code
+	// in the other three switch branches is very repetitive
+	if req.GetOutFromResource() == nil {
+		labels = append(labels, promLabel("namespace", req.Resource.Spec.Namespace))
+		if req.Resource.Spec.Name != "" {
+			labels = append(labels, promLabel(req.Resource.Spec.Type, req.Resource.Spec.Name))
+		}
+	}
+	labels = append(labels, promLabel("direction", direction))
+
+	return strings.Join(labels, ",")
+}
+
+func (h *handler) getRequests(ctx context.Context, reqLabels string, groupBy string, timeWindow string) (map[string]*pb.BasicStats, error) {
+	requestsQuery := fmt.Sprintf(reqQuery, reqLabels, timeWindow, groupBy)
+
+	resultVector, err := h.queryProm(ctx, requestsQuery)
+	if err != nil {
+		return nil, err
+	}
+	return processRequests(resultVector, groupBy), nil
+}
+
+func processRequests(vec model.Vector, labelSelector string) map[string]*pb.BasicStats {
+	result := make(map[string]*pb.BasicStats)
+	for _, sample := range vec {
+		labels := metricToMap(sample.Metric)
+		if result[labels[labelSelector]] == nil {
+			result[labels[labelSelector]] = &pb.BasicStats{}
+		}
+
+		switch labels["classification"] {
+		case "success":
+			result[labels[labelSelector]].SuccessCount = uint64(sample.Value)
+		case "fail":
+			result[labels[labelSelector]].FailureCount = uint64(sample.Value)
+		}
+	}
+	return result
+}
+
+func metricToMap(metric model.Metric) map[string]string {
+	labels := make(map[string]string)
+	for k, v := range metric {
+		labels[string(k)] = string(v)
+	}
+	return labels
+}
+
+func (h *handler) getDeployment(namespace string, name string) ([]appsv1.Deployment, map[string]*meshedCount, error) {
+	if namespace == "" {
+		namespace = apiv1.NamespaceDefault
+	}
+
+	deploymentsClient := h.k8sClient.Apps().Deployments(namespace)
+	deployment, err := deploymentsClient.Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	meshCount, err := h.getMeshedPodCount(namespace, deployment)
+	if err != nil {
+		return nil, nil, err
+	}
+	meshMap := map[string]*meshedCount{deployment.Name: meshCount}
+	return []appsv1.Deployment{*deployment}, meshMap, nil
+}
+
+func (h *handler) getDeployments(namespace string) ([]appsv1.Deployment, map[string]*meshedCount, error) {
```

Comment:
is it possible/advisable to implement `getDeployment` and `getDeployments` in terms of each other?